war-wrapper/src/main/java/org/ow2/proactive/scheduler/util/WarWrapper.java

```diff
@@ -0,0 +1,217 @@
+
+        } catch (Exception e) {
+            logger.error("Error when starting the scheduler", e);
+        }
+
+    }
+
+    protected void configureSchedulerAndRMAndPAHomes() {
+
+        String schedHome = getClass().getProtectionDomain().getCodeSource().getLocation().getPath() +
+                           SCHEDULER_RM_DEFAULT_HOME;
+
+        setPropIfNotAlreadySet(PASchedulerProperties.SCHEDULER_HOME.getKey(), schedHome);
+        schedHome = System.getProperty(PASchedulerProperties.SCHEDULER_HOME.getKey());
+
+        setPropIfNotAlreadySet(PASchedulerProperties.SCHEDULER_HOME.getKey(), schedHome);
+        setPropIfNotAlreadySet(PAResourceManagerProperties.RM_HOME.getKey(), schedHome);
+        setPropIfNotAlreadySet(CentralPAPropertyRepository.PA_HOME.getName(), schedHome);
+        setPropIfNotAlreadySet(CentralPAPropertyRepository.PA_CONFIGURATION_FILE.getName(),
+                               schedHome + "/config/network/server.ini");
+    }
+
+    //**************************************Shutting down ProActive ***************************************************
+    public void shutdownProactive() {
+
+        try {
+
+            Credentials credentials = getCredentials();
+
+            if (stopScheduler(credentials))
+                logger.info("Scheduler on " + schedulerUri + " correctly stopped");
+            else
+                logger.warn("Scheduler on " + schedulerUri + " is not stopped");
+
+            if (stopRM(credentials))
+                logger.info("RM on " + rmURL + " correctly stopped");
+            else
+                logger.warn("RM on " + rmURL + " is not stopped");
+
+        } catch (KeyException e) {
+            logger.error("ERROR while stopping ProActive: Cannot  acquire credentials");
+        } catch (Exception e) {
+            logger.error("ERROR while stopping ProActive: " + e.getMessage());
+        }
+    }
+
+    protected boolean stopRM(Credentials credentials) {
+
+        boolean rmStopped = false;
+
+        try {
+            RMAuthentication rmAuthentication = connectToRM();
+            ResourceManager rm = rmAuthentication.login(credentials);
+
+            //Do not kill Runtime when stopping RM
+            PAResourceManagerProperties.RM_SHUTDOWN_KILL_RUNTIME.updateProperty("false");
+
+            BooleanWrapper booleanWrapper = rm.shutdown(true);
+            rmStopped = booleanWrapper.getBooleanValue();
+
+        } catch (Exception e) {
+            logger.error("ERROR while stopping RM on " + rmURL + ": " + e.getMessage());
+        }
+
+        return rmStopped;
+    }
+
+    protected boolean stopScheduler(Credentials credentials) {
+
+        boolean schedulerStopped = false;
+
+        try {
+
+            Scheduler scheduler = schedAuthInter.login(credentials);
+            schedulerStopped = scheduler.shutdown();
+
+            if (schedulerStopped && scheduler.getStatus().isDown())
+                SchedulerFactory.schedulerStarted = false;
```

Comment:
I think using a static public property here is not really elegant. I'm sure there is a cleaner way